### ISSUE TITLE
isValid throws console.warn when pass the string

### ISSUE THREAD

**Issue by tahirdemirtas** [date-fns/date-fns#1575]
I'm trying to check the date is valid or not with isValid and it returns false when I pass the string eg. '2016-01-01' But toDate function is still logging the two console warnings

First one:
`Starting with v2.0.0-beta.1 date-fns doesn't accept strings as arguments. Please use "parseISO" to parse strings. See: https://git.io/fjule`

Second one:
`Error
    at toDate (index.js:50)
    at isValid (index.js:68)
...
`

I don't know it's intended or not but IMHO toDate should not throw warnings when isValid used.

**Comment by Imballinst** [comment]
Let me see if we are on the same page:

1. You are using `isValid` with string arguments.
2. `isValid` calls `toDate` in the function (see https://github.com/date-fns/date-fns/blob/master/src/isValid/index.js#L64)
3. When the argument is a string, it will log out a warning just like what you reported (see https://github.com/date-fns/date-fns/blob/6a99c2eb5e96fd6b20d245a9e26cbe4a6adbd2b6/src/toDate/index.js#L52-L57)

I think that's intended (?) You should `parseISO` first before passing it to `isValid`. Or, give a number (in milliseconds) would do, too. Let me know if this helps!

**Comment by Fortidude** [comment]
Well.. passing it with parseISO makes no sens...

https://date-fns.org/v2.9.0/docs/isValid


Name | Type | Description
-- | -- | --
date | * | the date to check

you should change the documentation, or remove the error and just return false. Ofcourse in my opinion.

**Comment by Imballinst** [comment]
@Fortidude good catch. I think what you said makes sense. In that case, I think this is the line where we would want to change the `*` to `Date | Number`?

https://github.com/date-fns/date-fns/blob/62eecac8247f33dd77e059803191811a3519cc6d/src/isValid/index.js#L42

**Comment by Fortidude** [comment]
Yes, changing the `*` to `Date | Number` is one of the solution 🤔   Currently it's a bit confusing - `isValid` could also return false for any type different than `Date | Number`.

**Comment by Imballinst** [comment]
> Currently it's a bit confusing - `isValid` could also return false for any type different than `Date | Number`.

apologies if I'm asking the obvious -- could you go again on this part? I thought this is intended since we can only parse `Date | Number`, so other types are not valid. Therefore, it returns `false`.

**Comment by ropinheiro** [comment]
I should also add to this that maybe date-fns should throw exceptions instead of polluting the console with warnings. I'm doing some code that try-catch a format() attempt, as I can't be sure if the data I receive is a valid date or not (it can be a valid date, or a string with a valid date formatted, or a string that must pass as-is), then in the catch I handle the invalid cases. The problem is that, although my code works pretty well, date-fns constantly fills the console with huge dumps each time format function isn't able to format the date. The catch prevents the exception to be written in the console, but the format function writing warnings escapes to my control. Isn't there a "quiet" option to make it behave silently?
